src/Xamarin.Android.Build.Tasks/Utilities/LlvmIrGenerator/LlvmIrGenerator.cs

```diff
@@ -923,19 +993,31 @@ public void WriteNameValueArray (string symbolName, IDictionary<string, string>
 				WriteArrayString (value, $"v_{i}");
 				i++;
 			}
+
 			if (strings.Count > 0) {
 				Output.WriteLine ();
 			}
 
-			WriteGlobalSymbolStart (symbolName, LlvmIrVariableOptions.GlobalConstantStringPointer);
+			WriteStringArray (symbolName, LlvmIrVariableOptions.GlobalConstantStringPointer, strings);
+
+			void WriteArrayString (string str, string symbolSuffix)
+			{
+				string name = WriteUniqueString ($"__{symbolName}_{symbolSuffix}", str, ref arrayStringCounter, LlvmIrVariableOptions.LocalConstexprString, out ulong size);
```

Comment:
Why not have `WriteUniqueString()` return a `StringSymbolInfo` instance?  This would allow it to drop the `out ulong size` parameter as well.